src/utilities/findDescriptionChanges.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-restricted-syntax */
+// @flow
+
+import {
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  GraphQLEnumType,
+  GraphQLInputObjectType,
+} from '../type/definition';
+import type { GraphQLFieldMap } from '../type/definition';
+import { GraphQLSchema } from '../type/schema';
+import invariant from '../jsutils/invariant';
+
+export const DescribedObjectType = {
+  FIELD: 'FIELD',
+  TYPE: 'TYPE',
+  ARGUMENT: 'ARGUMENT',
+  ENUM_VALUE: 'ENUM_VALUE',
+};
+
+export const DescriptionChangeType = {
+  OBJECT_ADDED: 'OBJECT_ADDED',
+  DESCRIPTION_ADDED: 'DESCRIPTION_ADDED',
+  DESCRIPTION_CHANGED: 'DESCRIPTION_CHANGED',
+};
+
+export type DescriptionChange = {
+  object: $Keys<typeof DescribedObjectType>,
+  change: $Keys<typeof DescriptionChangeType>,
+  description: string,
+  oldThing: any,
+  newThing: any,
+};
+
+/**
+ * Given two schemas, returns an Array containing descriptions of any
+ * descriptions that are new or changed and need review.
+ */
+export function findDescriptionChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema,
+): Array<DescriptionChange> {
+  const oldTypeMap = oldSchema.getTypeMap();
+  const newTypeMap = newSchema.getTypeMap();
+
+  const descriptionChanges: Array<?DescriptionChange> = [];
+
+  Object.keys(newTypeMap).forEach(typeName => {
+    const oldType = oldTypeMap[typeName];
+    const newType = newTypeMap[typeName];
+
+    descriptionChanges.push(
+      generateDescriptionChange(newType, oldType, DescribedObjectType.TYPE),
+    );
+
+    if (
+      newType instanceof GraphQLObjectType ||
+      newType instanceof GraphQLInterfaceType ||
+      newType instanceof GraphQLInputObjectType
+    ) {
+      invariant(
+        !oldType ||
+          oldType instanceof GraphQLObjectType ||
+          oldType instanceof GraphQLInterfaceType ||
+          oldType instanceof GraphQLInputObjectType,
+        'Expected oldType to also have fields',
```

Comment:
It shouldn't be an error.
For example:
```graphql
scalar Date
```
```graphql
type Date {
  day: Int
  month: Int
  year: Int
}
```

Reply:
This was added to satisfy Flow (https://github.com/graphql/graphql-js/pull/1127#discussion_r155619374).

This (and the enum case) are breaking schema changes, so an error seemed appropriate? Would you prefer they were silently ignored? If so, how do I do that in a way that Flow will recognize?

Reply:
So if I do breaking change in my schema I don't need to notify my documentation team about other non-breaking changes in my schema.
BTW Breaking changes is normal in many cases for example if you have your API in public demo.

Reply:
Yes, you're right. I was thinking too much about our specific case (where breaking changes already ping a substantial set of people for review and the docs changes are less relevant) but in the general case this should absolutely work.

I'm still not sure how to satisfy Flow without an invariant, but that's purely me not being much of a javascript developer.